Source/santad/SNTEventLog.m

```diff
@@ -501,4 +508,50 @@ - (NSString *)serialForDevice:(NSString *)devPath {
   return serial;
 }
 
+/**
+ Uses the executable path, uid, and gid from a given santa_message_t to determine if the path
+ has been translocated by GateKeeper and if so, returns the original path of the executable.  This
+ requires macOS 10.12 or higher.  We use dlopen to access the functions we need in
+ Security.framework so that we can still build against the 10.11 SDK.  If the path has not been
+ translocated or if running on macOS prior to 10.12, this method returns nil.
+ */
+- (NSString *)originalPathForTranslocation:(santa_message_t)message {
+  // The first time this function is called, we attempt to find the addresses of
+  // SecTranslocateIsTranslocatedURL and SecTranslocateCreateOriginalPathForURL inside of the
+  // Security.framework library.  If we were successful, handle will be non-NULL and is never
+  // closed.
+  static Boolean (*IsTranslocatedURL)(CFURLRef, bool *, CFErrorRef *) = NULL;
+  static CFURLRef __nullable (*CreateOriginalPathForURL)(CFURLRef, CFErrorRef *) = NULL;
+  static dispatch_once_t token;
+  dispatch_once(&token, ^{
+    void *handle = dlopen("/System/Library/Frameworks/Security.framework/Security", RTLD_LAZY);
+    if (handle) {
+      IsTranslocatedURL = dlsym(handle, "SecTranslocateIsTranslocatedURL");
+      CreateOriginalPathForURL = dlsym(handle, "SecTranslocateCreateOriginalPathForURL");
+      if (!IsTranslocatedURL || !CreateOriginalPathForURL) {
+        IsTranslocatedURL = NULL;
+        CreateOriginalPathForURL = NULL;
+        dlclose(handle);
+      }
+    }
+  });
+
+  // If we couldn't open the library or find the functions we need, don't do anything.
+  if (!IsTranslocatedURL || !CreateOriginalPathForURL) return nil;
+
+  // Determine if the executable URL has been translocated or not.
+  CFURLRef cfExecURL = (__bridge CFURLRef)[NSURL fileURLWithPath:@(message.path)];
+  bool isTranslocated = false;
+  CFErrorRef cferror = NULL;
```

Comment:
You might as well forego this actually as you're never reading the error, you can instead just pass NULL where cferror is being used.